src/private/files.ts

```diff
@@ -258,4 +274,63 @@ export namespace files {
 
     sendMessageToParent('files.openCloudStorageFile', [file, providerCode, fileOpenPreference]);
   }
+
+  /**
+   * @private
+   * Allow 1st party apps to call this function to get the external
+   * third party cloud storage accounts that the tenant supports
+   * @param excludeAddedProviders: return a list of support third party
+   * cloud storages that hasn't been added yet.
+   */
+  export function getExternalProviders(
+    excludeAddedProviders = false,
+    callback: (error: SdkError, providers: IExternalProvider[]) => void,
+  ): void {
+    ensureInitialized();
+
+    if (!callback) {
+      throw new Error('[files.getExternalProviders] Callback cannot be null');
+    }
+
+    sendMessageToParent('files.getExternalProviders', [excludeAddedProviders], callback);
+  }
+
+  /**
+   * @private
+   * Allow 1st party apps to call this function to move files
+   * among SharePoint and third party cloud storages.
+   */
+  export function copyMoveFiles(
```

Comment:
nit: `copyMoveFiles` for an API name might be confusing if we chose to make this public.
Can we have `copyFiles` and `moveFiles` as separate functions? Then no need to ask for `isMove` param.

Reply:
Then can we at least rename it to be `copyOrMoveFiles`?